File: all/abc212/abc212_b/abc212_b.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
  string ans = "Strong";
  char x[4];
  int count = 0;
  for(int i = 0; i < 4; i++) cin >> x[i];
  if(x[0] == x[1] && x[1] == x[2] && x[2] == x[3]) ans = "Weak";
  else {
    for(int i = 0; i < 3; i++) {
      if(x[i] == '9') {
        if(x[i+1] == '0') count++;
      } else {
        if((int)x[i] == (int)x[i+1] - 1) count++;
      } 
    }
  }
  if(count == 3) ans = "Weak";
  cout << ans;
}

File: abc/abc212/abc212_b/abc212_b.py
s = input()

if int(s) % 1111 == 0:
  print('Weak')
  exit()

for i in range(3):
  if ((int(s[i]) + 1) % 10) != int(s[i + 1]): break
  if i == 2:
    print('Weak')
    exit()

print('Strong')